sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ImplementationBridgeHelpers.java

```diff
@@ -55,27 +54,30 @@
 public class ImplementationBridgeHelpers {
     private final static Logger logger = LoggerFactory.getLogger(ImplementationBridgeHelpers.class);
     public static final class CosmosClientBuilderHelper {
-        private static CosmosClientBuilderAccessor accessor;
+        private static final AtomicReference<CosmosClientBuilderAccessor> accessor = new AtomicReference<>();
+        private static final AtomicBoolean cosmosClientBuilderClassLoaded = new AtomicBoolean(false);
 
         private CosmosClientBuilderHelper() {}
-        static {
-            ensureClassLoaded(CosmosClientBuilder.class);
-        }
 
         public static void setCosmosClientBuilderAccessor(final CosmosClientBuilderAccessor newAccessor) {
-            if (accessor != null) {
-                throw new IllegalStateException("CosmosClientBuilder accessor already initialized!");
+            if (!accessor.compareAndSet(null, newAccessor)) {
```

Comment:
should we call doNothingButEnsureLoadingClass() in setter as well?

Reply:
No - because setXXX will always be called from the class that we want to ensure to be loaded.

Reply:
oh yea make sense